### DISCUSSION TITLE
Support for deeply-nested included resources?

### DISCUSSION THREAD

**Discussion by esmail** [django-json-api/django-rest-framework-json-api#957] [Q&A] [answered]
Taking the familiar example of a blog post/article, would it be possible to return both the comments on the article *and* the authors of those comments?

The following quote (emphasis mine) leads me to believe this use case to be in-spec., but I'm not sure how to do so with DJA:
> Note: Full linkage ensures that included resources are related to either the primary data (which could be resource objects or resource identifier objects) *or to each other*.

https://jsonapi.org/format/#document-compound-documents

**Comment by sliverc** [comment, marked as answer]
Yes this is possible as per [specification](https://jsonapi.org/format/#fetching-includes) include needs to be defined dotted.

So for your example the include query parameter should look like this:

```
GET /articles/1?include=comments.author HTTP/1.1
```

For this to work you define `included_serializers` in your serializers as per [documentation](https://django-rest-framework-json-api.readthedocs.io/en/stable/usage.html#included).

**Reply by esmail** [reply]
Forgot to say thanks for this. I was mistakenly thinking I needed to somehow reference `author` in `MySerializer.Meta.fields`. I'll try to circle back and PR in a quick mention of the in the docs.
